akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1393,33 +1500,104 @@ private[cluster] final class FirstSeedNodeProcess(seedNodes: immutable.IndexedSe
+              "'akka.cluster.configuration-compatibility-check.enforce-on-join = off'. " +
+              "Note that disabling it will allow the formation of a cluster with nodes having incompatible configuration settings. " +
+              "This node will be shutdown!")
+          CoordinatedShutdown(context.system).run(IncompatibleConfigurationDetected)
+      }
+
+    case InitJoinAck(address, UncheckedConfig) ⇒
+      // first InitJoinAck reply, but with unchecked configuration
+      if (ByPassConfigCompatCheck) {
+        log.info("Received InitJoinAck message from [{}] to [{}]", sender(), selfAddress)
+        log.warning("Joining cluster without configuration compatibility check feature. " +
+          "Join will be performed because compatibility check is configured to not be enforced.")
+        // only join if set to ignore config validation
+        context.parent ! JoinTo(address)
+        context.become(done)
+      } else {
+        log.error(
+          "Couldn't join seed nodes because the cluster does NOT support configuration compatibility checks. " +
+            "It's recommended to perform a full cluster shutdown in order to deploy this new version." +
+            "If a cluster shutdown isn't an option, you may want to disable this protection by setting " +
+            "'akka.cluster.configuration-compatibility-check.enforce-on-join = off'. " +
+            "Note that disabling it will allow the formation of a cluster with nodes having incompatible configuration settings. " +
+            "This node will be shutdown!")
+        CoordinatedShutdown(context.system).run(IncompatibleConfigurationDetected)
+      }
+
+    case InitJoinAck(address, IncompatibleConfig) ⇒
+      // first InitJoinAck reply, but incompatible
+      if (ByPassConfigCompatCheck) {
+        log.info("Received InitJoinAck message from [{}] to [{}]", sender(), selfAddress)
+        log.warning("Joining cluster with incompatible configurations. " +
+          "Join will be performed because compatibility check is configured to not be enforced.")
+        // only join if set to ignore config validation
+        context.parent ! JoinTo(address)
+        context.become(done)
+      } else {
+        log.error(
+          "Couldn't join seed nodes because of incompatible cluster configuration. " +
+            "It's recommended to perform a full cluster shutdown in order to deploy this new version." +
+            "If a cluster shutdown isn't an option, you may want to disable this protection by setting " +
+            "'akka.cluster.configuration-compatibility-check.enforce-on-join = off'. " +
+            "Note that disabling it will allow the formation of a cluster with nodes having incompatible configuration settings. " +
+            "This node will be shutdown!")
+        CoordinatedShutdown(context.system).run(IncompatibleConfigurationDetected)
+      }
```

Comment:
could we reduce the duplicated code for these checks in `FirstSeedNodeProcess` and `JoinSeedNodeProcess`?